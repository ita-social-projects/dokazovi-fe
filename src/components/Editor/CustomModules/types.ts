import Quill from 'quill';
import { DropEvent, FileRejection } from 'react-dropzone';

export interface IUrlInputModalProps {
  editor?: Quill;
  updateBackgroundImage?: (url: string) => void;
  forBackgroundImg?: boolean;
}

export interface IBackgroundImageContainerProps {
  dispatchImageUrl: (backgroundImageUrl: string) => void;
  fileSelectorHandler: <T extends File>(
    acceptedFiles: T[],
    fileRejections: FileRejection[],
    event: DropEvent,
  ) => void;
  title?: string;
  imgUrl?: string;
  notCarousel?: boolean;
  reminder?: boolean;
  forMobilePic: boolean;
}

export interface IFileInputProps {
  onDrop: <T extends File>(
    acceptedFiles: T[],
    fileRejections: FileRejection[],
    event: DropEvent,
  ) => void;
  forMobilePic: boolean;
}
