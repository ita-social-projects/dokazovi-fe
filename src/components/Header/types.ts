export interface IHeaderProps {
  id: string;
  label: string;
  url: string;
}
