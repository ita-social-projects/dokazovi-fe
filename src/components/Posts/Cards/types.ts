import { IPost } from '../../../old/lib/types';

export interface IPostPreviewCardProps {
  post: IPost;
  size?: 'small' | 'large';
  shouldNotUseLink?: boolean;
}
