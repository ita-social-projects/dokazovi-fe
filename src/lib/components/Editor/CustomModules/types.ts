import Quill from 'quill';
import { CreatePostRequestType } from '../../../utilities/API/types';

export interface IUrlInputModalProps {
  editor?: Quill;
  updateBackgroundImage?: (url: string) => void;
}

export interface IBackgroundImageContainerProps {
  dispatchImageUrl: (backgroundImageUrl: string) => void;
  fileSelectorHandler: (e: React.ChangeEvent<HTMLInputElement>) => void;
  newPost?: CreatePostRequestType;
}
