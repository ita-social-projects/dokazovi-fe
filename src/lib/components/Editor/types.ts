import Quill from 'quill';

export interface IEditorToolbarProps {
  editor?: Quill;
}
