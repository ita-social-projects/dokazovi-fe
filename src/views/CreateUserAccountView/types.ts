export interface ICreateUserAccount {
  newPassword: string;
  matchPassword: string;
}
