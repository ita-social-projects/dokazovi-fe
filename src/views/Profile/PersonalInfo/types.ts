import { IExpert } from 'old/lib/types';

export interface INewAuthorValues {
  avatar?: string;
  firstName: string;
  lastName: string;
  regionId: number | null;
  cityId: number | null;
  bio: string;
  publicEmail?: string;
  socialNetworks: (string | null)[];
  mainWorkingPlace: string;
}

export interface IErrorFields {
  avatar: string;
  lastName: string;
  firstName: string;
  regionId: string;
  cityId: string;
  publicEmail: string;
  mainWorkingPlace: string;
  bio: string;
  socialNetworks: string[];
  socialNetwoksRequired: string;
}

export interface IVisitFields {
  lastName: boolean;
  firstName: boolean;
  region: boolean;
  city: boolean;
  publicEmail: boolean;
  mainWorkingPlace: boolean;
  bio: boolean;
  facebook: boolean;
  instagram: boolean;
  youtube: boolean;
  twitter: boolean;
  linkedin: boolean;
}

export interface IRegionCityHandlerProps {
  newAuthorValues: INewAuthorValues;
  visitFields: IVisitFields;
  errorMessages: IErrorFields;
  setNewAuthorValues: React.Dispatch<React.SetStateAction<INewAuthorValues>>;
  blurHandler: (
    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>,
  ) => void;
}

export interface IEditAuthorProps {
  author?: IExpert;
  isCurrentUser?: boolean;
  onSaveSuccessful: (id: number) => void;
}

export type UserEmailType = {
  publicEmail: (null | string)[];
  privateEmail: (null | string)[];
};

export type ActionButtonType = 'activate' | 'deactivate' | 'create' | 'nothing';
