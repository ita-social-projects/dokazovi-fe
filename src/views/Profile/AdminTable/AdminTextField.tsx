import React from 'react';
import TextField from '@material-ui/core/TextField';
import FormControl from '@material-ui/core/FormControl';
import { FieldEnum, IField } from '../../../models/adminLab/types';

interface IMaterialsTextField {
  value: string;
  field: FieldEnum;
  setChanges: (payload: IField) => void;
}

export const AdminTextField: React.FC<IMaterialsTextField> = ({
  value,
  field,
  setChanges,
}) => {
  const handleTextChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const text = event.target.value;
    setChanges({ field, text });
  };

  return (
    <FormControl>
      <TextField
        placeholder={field}
        value={value}
        onChange={handleTextChange}
      />
    </FormControl>
  );
};
