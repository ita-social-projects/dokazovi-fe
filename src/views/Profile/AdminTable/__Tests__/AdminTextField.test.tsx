import React from 'react';
import { render, screen } from '@testing-library/react';
import { AdminTextField } from '../AdminTextField';
import { FieldEnum } from '../../../../models/adminLab/types';

const mockOnSetChanged = jest.fn();

test('component renders properly with all props', () => {
  render(
    <AdminTextField
      field={FieldEnum}
      setChanges={mockOnSetChanged}
      inputValue="test"
      placeholder="test"
    />,
  );

  expect(screen.getByTestId('admin-text-filter')).toBeInTheDocument();
  expect(screen.getByTestId('admin-text-filter')).toMatchSnapshot();
});

test('component renders initial preview text', () => {
  render(
    <AdminTextField
      field={FieldEnum}
      setChanges={mockOnSetChanged}
      inputValue="test"
      placeholder="test"
    />,
  );

  expect(screen.getByPlaceholderText('test')).toBeInTheDocument();
});
