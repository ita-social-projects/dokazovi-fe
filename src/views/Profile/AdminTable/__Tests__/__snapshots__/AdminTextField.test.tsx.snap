// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders properly with all props 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root"
  data-testid="admin-text-filter"
>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input"
      placeholder="test"
      type="text"
      value="test"
    />
  </div>
</div>
`;
